File: 4. Java/EX23.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX23 
{
    public static void main(String[] args) 
    {
        // Formatação de números
        Locale.setDefault(Locale.US);
        
        // Leitura dos valores de X e Y
        try (Scanner scanner = new Scanner(System.in)) 
        {
            System.out.print("# Valor de X: ");
            double x = scanner.nextDouble();
            
            System.out.print("# Valor de Y: ");
            double y = scanner.nextDouble();
            
            // Exibe a posição no plano cartesiano
            System.out.println(determinePosition(x, y));
        }
    }

    // Determina a posição no plano cartesiano com base nos valores de X e Y
    private static String determinePosition(double x, double y)
    {
        if (x == 0 && y == 0) 
            return "# Origem";

        if (x == 0 && y != 0)
            return "# Eixo Y";

        if (x != 0 && y == 0) 
            return "# Eixo X";

        // Verifica os quadrantes
        if (x > 0)
            return (y > 0) ? "# Q1" : "# Q4";
        else
            return (y > 0) ? "# Q2" : "# Q3";
    }
}

File: 6. Python/EX23.py
# Solicita os valores de X e Y
x = float(input("# Valor de X: "))  # Lê o valor de X
y = float(input("# Valor de Y: "))  # Lê o valor de Y

# Verifica a posição de acordo com os eixos e quadrantes
if x == 0 and y == 0:
    print("# Origem")  # Ponto na origem

elif x == 0 and y != 0:
    print("# Eixo Y")  # Ponto no eixo Y

elif x != 0 and y == 0:
    print("# Eixo X")  # Ponto no eixo X

elif x > 0:
    if y > 0:
        print("# Q1")  # Primeiro quadrante
    else:
        print("# Q4")  # Quarto quadrante

elif x < 0:
    if y > 0:
        print("# Q2")  # Segundo quadrante
    else:
        print("# Q3")  # Terceiro quadrante
